bitnami/kafka/templates/scripts-configmap.yaml

```diff
@@ -72,7 +72,7 @@ data:
     #!/bin/bash
 
     ID="${MY_POD_NAME#"{{ $fullname }}-"}"
-    export KAFKA_CFG_BROKER_ID="$ID"
+    export KAFKA_CFG_BROKER_ID="$((ID + {{ .Values.minBrokerId }}))"
```

Comment:
Possible workaround to help upgrading charts when a different ID is set in the existing data:

```console
if [[ -f "/bitnami/kafka/data/meta.properties" ]]; then
    export KAFKA_CFG_BROKER_ID="$(grep "broker.id" /bitnami/kafka/data/meta.properties | awk -F '=' '{print $2}')"
else
    ID="${MY_POD_NAME#"{{ $fullname }}-"}"
    export KAFKA_CFG_BROKER_ID="$ID"
fi
```

Reply:
@juan131 but how can one populate `/bitnami/kafka/data/meta.properties`? or does it get parsed and populated from `/bitnami/kafka/config/server.properties` which in turn can be configured via`config:` in the chart?

Reply:
Id' try to explain myself. Let's say you install the chart with 3 brokers:

```console
$ helm install my-release bitnami/kafka --set replicaCount=3 --set minBrokerId=100
``` 

We can see each broker has its own PVC mounted at `/bitnami/kafka` to persist the Kafka data.

```console
$ kg pvc
NAME                     STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
data-kafka-0             Bound    pvc-14dc5cb4-83d7-41f4-b584-e4e0924c915e   8Gi        RWO            standard       69s
data-kafka-1             Bound    pvc-1c2539bf-0e02-4107-9879-8cbbe9eb04b0   8Gi        RWO            standard       69s
data-kafka-2             Bound    pvc-5b8dfb2b-8193-4721-ac82-1de9e360d962   8Gi        RWO            standard       69s
```

And... If we inspect the `meta.properties` the brokers:

```console
$ kubectl exec kafka-0 -- grep "broker.id" /bitnami/kafka/data/meta.properties
broker.id=100
$ kubectl exec kafka-1 -- grep "broker.id" /bitnami/kafka/data/meta.properties
broker.id=101
$ kubectl exec kafka-2 -- grep "broker.id" /bitnami/kafka/data/meta.properties
broker.id=102
```

Now.. let's say that you upgrade the chart (but you forget to set **minBrokerId**):

```console
$ helm upgrade kafka bitnami/kafka --set replicaCount=3
```

You'll find the error below:

```
$ kubectl logs kafka-2
...
kafka.common.InconsistentBrokerIdException: Configured broker.id 2 doesn't match stored broker.id 102 in meta.properties
```

With my proposal.. You avoid that error by ensuring you respect the broker id persisted in the volumes.

Reply:
@juan131 agree, just pushed the necessary changes.